### ISSUE TITLE
[Rule] Detect invalid FOR loop header

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#445]
For example:
```
FOR	${INDEX}	In RANGE	2	17
```
It will throw `has no IN or any valid separator` during execution but we could possibly detect it with Robocop (and we could check if RF have already some validations used in runtime we could copy over to our checks). 

Another possible error:
```
    FOR  val  IN RANGE  10
```
throws `Invalid FOR loop variable 'val'.` in runtime.

**Comment by bhirsz** [comment]
May be useful: https://github.com/robotframework/robotframework/blob/60f6ca42cd3657d5854b178eec1ce091fcbe3672/atest/testdata/running/for.robot

**Comment by bhirsz** [comment]
At least some of the issues are catched by parsing-error:
[E] 0401 Robot Framework syntax error: FOR loop has no 'IN' or other valid separator. (parsing-error)
